src/org/armedbear/lisp/Lisp.java

```diff
@@ -2143,12 +2143,10 @@ public static Pathname coerceToPathname(LispObject arg)
       return (Pathname) arg;
     if (arg instanceof AbstractString)
       return (Pathname)Pathname.create(((AbstractString)arg).toString());
-    if (arg instanceof FileStream)
```

Comment:
From visually inspecting things, I'm surprised that such elision for all the types works.  

The tests didn't complete 'cuz  gitlab.common-lisp.net was wedged for the ANSI-TEST suite.  I've fixed that, re-running tests now.  We'll see.

Reply:
gitlab.common-lisp.net seems to break a lot. Because Clasp's build system does lots of git cloning and has dependencies on gitlab.common-lisp.net, we've set up a mirror of some the repos. ansi-test is here https://github.com/clasp-developers/ansi-test if you want to use it. It refreshes every day.

Reply:
Added `@Override` for the various `getPathname` implementations.

Reply:
I've added tests to nontrivial-gray-streams for the other pathname components. They all pass for this PR.

https://github.com/yitzchak/nontrivial-gray-streams/blob/3275bb649636a962d13cd05498679bd231d538d4/code/test/macros.lisp#L57-L103

Reply:
Testing looks good wrt. ABCL's PATHNAME specializations (JAR-PATHNAME, URL-PATHNAME).  

Mebbe I will add NONTRIVIAL-GRAY-STREAMS to the CI's testing when I get a chance.